prototype2/gdgem/generators/nmxgen_hits.cpp

```diff
@@ -69,11 +77,21 @@ int main(int argc, char *argv[]) {
 
   size_t SentPackets{0}; // counter to determine when to break
   for (;;) {
-    int readsz = file.read(buffer);
-    if (readsz > 0 && (Settings.NumberOfPackets == 0 || SentPackets < Settings.NumberOfPackets)) {
-      DataSource.send(buffer, readsz);
-      Bytes += readsz;
-      Packets++;
+    if (Settings.NumberOfPackets == 0 || SentPackets < Settings.NumberOfPackets) {
+      int readsz = file.read(buffer);
```

Comment:
The declaration of `buffer` sets its size to 9000, which is compatible but has to be synchronised manually with the value of 9000 in _DumpFile.h:49_, should it ever change. Using a constant in this case could ensure the numbers stay the same.